### PR TITLE
enh(ci): Move Centreon Gorgone packaging to nfpm

### PR DESCRIPTION
**Pull request by hamzabessa**
## Description

**PLEASE MAKE SURE THAT THE BRANCH PR INCLUDES JIRA TICKET ID** (_for centreon-internal_)

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon-gorgone/packaging/centreon-gorgone.yaml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+homepage: "https://www.centreon.com"
+license: "Apache-2.0"
+
+contents:
+  - dst: "/etc/centreon-gorgone"
+    type: dir
+    file_info:
+      mode: 0775
+
+ - dst: "/centreon-gorgone/config.d"
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone.yaml:23]:
```suggestion
 - dst: "/etc/centreon-gorgone/config.d"
```

File: centreon-gorgone/packaging/centreon-gorgone.yaml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    type: dir
+    file_info:
+      mode: 0755
+
+ - dst: "/centreon-gorgone/config.d/cron.d"
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone.yaml:28]:
```suggestion
 - dst: "/etc/centreon-gorgone/config.d/cron.d"
```

File: centreon-gorgone/packaging/centreon-gorgone.yaml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  - src: "centreon-gorgone/config/systemd/gorgoned-sysconfig"
+    dst: "/etc/sysconfig/gorgoned"
+    type: config|noreplace
+
+  - src: "centreon-gorgone/config/logrotate/gorgoned"
+    dst: "/etc/logrotate.d/gorgoned"
+    type: config|noreplace
+
+  - src: "centreon-gorgone/gorgone/*"
+    dst: "/etc/${PERL_VENDORLIB}/gorgone/"
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone.yaml:47]:
```suggestion
    dst: "${PERL_VENDORLIB}/gorgone/"
```

File: centreon-gorgone/packaging/centreon-gorgone.yaml [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+  - src: "centreon-gorgone/packaging/centreon-audit.yaml"
+    dst: "/etc/centreon-gorgone/config.d/50-centreon-audit.yaml"
+    type: config|noreplace
+
+  - src: "centreon-gorgone/packaging/sudoers.d/centreon-gorgone"
+    dst: "/etc/sudoers.d/centreon-gorgone"
+    file_info:
+      mode: 0600
+      owner: root
+      group: root
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone.yaml:94]: should be copied in `centreon-gorgone-centreon-config.yaml` file

File: centreon-gorgone/packaging/centreon-gorgone-centreon-config.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+maintainer: "Centreon <contact@centreon.com>"
+description: |
+  Move Centreon Gorgone packaging to nfpm
+  Commit: @COMMIT_HASH@
+vendor: "Centreon"
+homepage: "https://www.centreon.com"
+license: "Apache-2.0"
+
+contents:
+  - src: "centreon-gorgone/packaging/configuration/centreon.yaml"
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone-centreon-config.yaml:18]:
```suggestion
  - src: "./configuration/centreon.yaml"
```

File: centreon-gorgone/packaging/scripts/centreon-gorgone-selinux-postinstall.sh [hunk 6]
```diff
@@ -0,0 +1 @@
+semodule -i %{_datadir}/selinux/packages/ent/ent-gorgoned.pp > /dev/null 2>&1
```
Comment (kduret) [inline centreon-gorgone/packaging/scripts/centreon-gorgone-selinux-postinstall.sh:1]: take example on https://github.com/centreon/centreon-collect/blob/develop/packaging/scripts/centreon-engine-selinux-postinstall.sh

File: .github/workflows/gorgone.yml [hunk 7]
```diff
@@ -60,21 +60,51 @@ jobs:
...
+
+      - name: Set package version and paths according to distrib
+        run: |
+          export VERSION="${{ needs.get-version.outputs.major_version }}"
+          if [ "${{ matrix.distrib }}" = "bullseye" ]; then
+            export PERL_VENDORLIB="/usr/share/perl5"
+          else
+            export VERSION="${{ needs.get-version.outputs.minor_version }}"
+            export PERL_VENDORLIB="/usr/share/perl5/vendor_perl"
+          fi
```
Comment (kduret) [inline .github/workflows/gorgone.yml:85]:
```suggestion
          VERSION="${{ needs.get-version.outputs.major_version }}"
          if [ "${{ matrix.distrib }}" = "bullseye" ]; then
            PERL_VENDORLIB="/usr/share/perl5"
          else
            VERSION="${{ needs.get-version.outputs.minor_version }}"
            PERL_VENDORLIB="/usr/share/perl5/vendor_perl"
          fi
          echo "VERSION=$VERSION" >> $GITHUB_ENV
          echo "PERL_VENDORLIB=$PERL_VENDORLIB" >> $GITHUB_ENV
```

File: centreon-gorgone/packaging/centreon-gorgone.yaml [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    type: dir
+    file_info:
+      mode: 0755
+
+  - src: "../config/systemd/gorgoned-service"
+    dst: "/etc/systemd/system/gorgoned.service"
+    file_info:
+      mode: 0755
+
+  - dst: "${PERL_VENDORLIB}/gorgone/"
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone.yaml:38]:
```suggestion
  - dst: "${PERL_VENDORLIB}/gorgone"
```

File: centreon-gorgone/packaging/centreon-gorgone.yaml [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+  - src: "../config/systemd/gorgoned-sysconfig"
+    dst: "/etc/sysconfig/gorgoned"
+    type: config|noreplace
+
+  - src: "../config/logrotate/gorgoned"
+    dst: "/etc/logrotate.d/gorgoned"
+    type: config|noreplace
+
+  - src: "../gorgone/*"
+    dst: "${PERL_VENDORLIB}/gorgone/"
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone.yaml:52]:
```suggestion
  - src: "../gorgone"
    dst: "${PERL_VENDORLIB}/gorgone"
```

File: centreon-gorgone/packaging/centreon-gorgone.yaml [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+
+  - src: "../gorgoned"
+    dst: "usr/bin/"
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone.yaml:55]:
```suggestion
    dst: "/usr/bin/"
```

File: .github/workflows/gorgone.yml [hunk 11]
```diff
@@ -60,21 +60,59 @@ jobs:
...
+          echo "VERSION=$VERSION"
+          echo "PERL_VENDORLIB=$PERL_VENDORLIB"
+        shell: bash
+
+      - name: Package
+        uses: ./.github/actions/package-nfpm
+        with:
+          nfpm_file_pattern: "centreon-gorgone/packaging/*.yaml"
+          distrib: ${{ matrix.distrib }}
+          package_extension: ${{ matrix.package_extension }}
```
Comment (kduret) [inline .github/workflows/gorgone.yml:101]:
```suggestion
          package_extension: ${{ matrix.package_extension }}
          major_version: ${{ needs.get-version.outputs.major_version }}
          minor_version: ${{ needs.get-version.outputs.minor_version }}
```

File: .github/workflows/gorgone.yml [hunk 12]
```diff
@@ -60,21 +60,59 @@ jobs:
...
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Set package version and paths according to distrib
+        run: |
+          VERSION="${{ needs.get-version.outputs.major_version }}"
+          if [ "${{ matrix.distrib }}" = "bullseye" ]; then
+            PERL_VENDORLIB="/usr/share/perl5"
+          else
+            VERSION="${{ needs.get-version.outputs.minor_version }}"
```
Comment (kduret) [inline .github/workflows/gorgone.yml:83]:
```suggestion
```

File: .github/workflows/gorgone.yml [hunk 13]
```diff
@@ -60,21 +60,59 @@ jobs:
...
+            PERL_VENDORLIB="/usr/share/perl5/vendor_perl"
+          fi
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          echo "PERL_VENDORLIB=$PERL_VENDORLIB" >> $GITHUB_ENV
+        shell: bash
+
+      - name: check environment variables are used correctly
+        run: |
+          echo "VERSION=$VERSION"
```
Comment (kduret) [inline .github/workflows/gorgone.yml:92]:
```suggestion
```

File: .github/workflows/gorgone.yml [hunk 14]
```diff
@@ -60,21 +60,57 @@ jobs:
...
+        shell: bash
+
+      - name: Package
+        uses: ./.github/actions/package-nfpm
+        with:
+          nfpm_file_pattern: "centreon-gorgone/packaging/*.yaml"
+          distrib: ${{ matrix.distrib }}
+          package_extension: ${{ matrix.package_extension }}
+          major_version: ${{ needs.get-version.outputs.major_version }}
+          minor_version: ${{ needs.get-version.outputs.minor_version }}
```
Comment (kduret) [inline .github/workflows/gorgone.yml:99]:
```suggestion
          minor_version: ${{ needs.get-version.outputs.minor_version }}
          release: 1
```

File: centreon-gorgone/packaging/centreon-gorgone-selinux.yaml [hunk 15]
```diff
@@ -0,0 +1,35 @@
+name: "centreon-gorgone"
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone-selinux.yaml:1]:
```suggestion
name: "centreon-gorgone-selinux"
```

File: .github/workflows/gorgone.yml [hunk 16]
```diff
@@ -60,21 +60,73 @@ jobs:
...
+          cd centreon-gorgone/selinux
+          MODULE="centreon-gorgoned"
+          sed -i "s/@VERSION@/${{ needs.get-version.outputs.major_version }}.${{ needs.get-version.outputs.minor_version }}/g" $MODULE.te
+          make -f /usr/share/selinux/devel/Makefile
+          cd -
+        shell: bash
+
+      - name: Remove selinux packaging files on debian
+        if: ${{ matrix.package_extension == 'deb' }}
+        run: rm -f "./centreon-gorgone/packaging/*-selinux.yaml"
```
Comment (kduret) [inline .github/workflows/gorgone.yml:104]:
```suggestion
        run: rm -f packaging/*-selinux.yaml
```

File: .github/workflows/gorgone.yml [hunk 17]
```diff
@@ -60,21 +60,73 @@ jobs:
...
+            PERL_VENDORLIB="/usr/share/perl5/vendor_perl"
+          fi
+          echo "PERL_VENDORLIB=$PERL_VENDORLIB" >> $GITHUB_ENV
+        shell: bash
+
+      - name: check environment variables are used correctly
+        run: |
+          echo "PERL_VENDORLIB=$PERL_VENDORLIB"
+        shell: bash
+
```
Comment (kduret) [inline .github/workflows/gorgone.yml:91]:
```suggestion
```

File: .github/workflows/gorgone.yml [hunk 18]
```diff
@@ -60,21 +60,68 @@ jobs:
...
+          cd centreon-gorgone/selinux
+          MODULE="centreon-gorgoned"
+          sed -i "s/@VERSION@/${{ needs.get-version.outputs.major_version }}.${{ needs.get-version.outputs.minor_version }}/g" $MODULE.te
+          make -f /usr/share/selinux/devel/Makefile
+          cd -
+        shell: bash
+
+      - name: Remove selinux packaging files on debian
+        if: ${{ matrix.package_extension == 'deb' }}
+        run: rm -f packaging/*-selinux.yaml
```
Comment (kduret) [inline .github/workflows/gorgone.yml:99]:
```suggestion
        run: |
          rm -f packaging/*-selinux.yaml
          pwd
          ls
          echo "after ls"
          ls packaging
```

File: .github/workflows/gorgone.yml [hunk 19]
```diff
@@ -60,21 +60,73 @@ jobs:
...
+        shell: bash
+
+      - name: Remove selinux packaging files on debian
+        if: ${{ matrix.package_extension == 'deb' }}
+        run: |
+          rm -f packaging/*-selinux.yaml
+          pwd
+          ls
+          echo "after ls"
+          ls packaging
```
Comment (kduret) [inline .github/workflows/gorgone.yml:104]:
```suggestion
          rm -f centreon-gorgone/packaging/*-selinux.yaml
          pwd
          ls centreon-gorgone
          echo "after ls"
          ls centreon-gorgone/packaging
```

File: .github/workflows/gorgone.yml [hunk 20]
```diff
@@ -60,21 +60,73 @@ jobs:
...
+        shell: bash
+
+      - name: Remove selinux packaging files on debian
+        if: ${{ matrix.package_extension == 'deb' }}
+        run: |
+          rm -f centreon-gorgone/packaging/*-selinux.yaml
+          pwd
+          ls centreon-gorgone
+          echo "after ls"
+          ls centreon-gorgone/packaging
```
Comment (kduret) [inline .github/workflows/gorgone.yml:104]:
```suggestion
        run: rm -f centreon-gorgone/packaging/*-selinux.yaml
```

File: centreon-gorgone/packaging/centreon-gorgone-selinux.yaml (hunk 15 above)
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone-selinux.yaml:1]:
```suggestion
name: "centreon-gorgone-selinux"
```

File: centreon-gorgone/packaging/centreon-gorgone-selinux.yaml [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+  Selinux for centreon-gorgone
+  Commit: @COMMIT_HASH@
+vendor: "Centreon"
+homepage: "https://www.centreon.com"
+license: "Apache-2.0"
+
+contents:
+  - dst: "/usr/share/selinux/packages/centreon"
+    type: dir
+
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone-selinux.yaml:20]: created by centreon-common-selinux package
```suggestion
```

File: centreon-gorgone/packaging/centreon-gorgone.yaml [hunk 22]
```diff
@@ -0,0 +1,151 @@
+name: "centreon-gorgone-selinux"
```
Comment (kduret) [inline centreon-gorgone/packaging/centreon-gorgone.yaml:1]:
```suggestion
name: "centreon-gorgone"
```

File: centreon-gorgone/config/systemd/gorgoned.deb.service [hunk 23]
```diff
@@ -0,0 +1,33 @@
+##
+## Copyright 2019-2020 Centreon
```
Comment (sc979) [inline centreon-gorgone/config/systemd/gorgoned.deb.service:2]:
```suggestion
## Copyright 2019-2023 Centreon
```

**Review by sc979 — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Pull request merged**